### DISCUSSION TITLE
Calculate depth integrated transport from meridional velocity along a cross section

### DISCUSSION THREAD

**Discussion by pecos27** [pydata/xarray#6041] [Q&A] [answered]
How might one calculate transport with xarray?
I am hoping to calculate depth integrated transports along a cross section with time, depth, and distance. The variable being meridional velocity.

The transport is defined as :

transport = meridional_velocity*A,

where A = cross sectional area = dx * dz, where dx is grid size along distance and dz is the grid size along depth, where the grid size in both distance and depth is 1/20th of a degree. 

I am hoping to get an output of the mean transport with respect to distance, so just a time series with transport on the y axis and distance on the x axis. What might be the best way to achieve this in xarray?

Apologies that this question is not accompanied with a reproducible example, but this is just more of a general question. If anyone may have some tips on formatting or example code on calculating depth integrated velocity transport, anything would be appreciated. Thanks!

**Comment by nmalan** [comment, marked as answer]
I would direct you towards xgcm (https://xgcm.readthedocs.io/en/latest/) as the right tool for this problem. Once you set up the grid, it will enable you to do your calculations while taking into account the size of each grid cell.
